src/scene_worker.js

```diff
@@ -44,8 +49,21 @@ if (Utils.isWorkerThread) {
         config = JSON.parse(config);
 
         for (var name in config.sources) {
-            let source = config.sources[name];
-            SceneWorker.sources[name] = TileSource.create(Object.assign(source, {name}));
+            let source = DataSource.create(Object.assign(config.sources[name], {name}));
+            if (source.tiled) {
+                SceneWorker.sources.tiles[name] = source;
+            }
+            else {
+                // Distribute object sources across workers
+                if (source.id % SceneWorker.num_workers === SceneWorker.worker_id) {
```

Comment:
Thinking about this now, might make more sense to do this filter on the main thread and never send sources to the workers that they don't need.